packages/jest-jasmine2/src/index.js

```diff
@@ -85,12 +92,24 @@ async function jasmine2(
     if (config.resetMocks) {
       runtime.resetAllMocks();
 
+      if (!environment) {
+        throw new ReferenceError(
+          'The environment reference that Jasmine has was cleaned too early. ' +
+            'Reference should only be cleaned after all tests have finished',
+        );
+      }
+
       if (config.timers === 'fake') {
         environment.fakeTimers.useFakeTimers();
       }
     }
   });
 
+  // Free references to environment to avoid leaks.
+  env.afterAll(() => {
+    environment = null;
```

Comment:
Who is holding on to `environment`?

Reply:
The `beforeAll` closure, through `environment.fakeTimers.useFakeTimers()`.